iceoryx_utils/testutils/timing_test.hpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <atomic>
+#include <cstdint>
+#include <functional>
+#include <string>
+
+/// @brief This headers provides TIMING_TEST unit test infrastructure. The idea
```

Comment:
The name "TIMING_TEST" is a litle bit misleading here because there are no timing constrints checked here. What is here done is to repeat the test until the condition is met or to fail. I know the background why we are doing that but then i would rename it to something like "REPEAT_TEST". That name is more meaningful. Iknow that we have different timing constraints on different platforms like arm or x64. But this should be tested different.

Maybe we can use the timeout mechanism of the semaphore to offer a real timing test which checks for a certain timing constraint?

Reply:
I think we should stick to the name TIMING_TEST since with this name we communicate to fellow developers that we are testing a construct with timing constraints. Our approach is to test it repeatedly but maybe later we have an even better approach and then the name should be the same and the rest should just change under the hood.